src/SizeWatcherProvider.js

```diff
@@ -16,19 +16,34 @@ export default class SizeWatcherProvider extends Component {
   }
 
   componentDidMount() {
+    const {sync} = this.props;
+    let raf;
```

Comment:
@chbdetta I think to be on safe side we still need to cancel in unmount, os it's should be on `this`